orca-spring-batch/src/main/groovy/com/netflix/spinnaker/orca/batch/RetryableTaskTasklet.groovy

```diff
@@ -91,15 +91,18 @@ class RetryableTaskTasklet extends TaskTasklet {
     def executionTimeMs = determineCurrentExecutionTime(clock, startTime, stage.execution.paused)
     def timeoutMs = (stage.context[STAGE_TIMEOUT_OVERRIDE_KEY] ?: timeoutMs) as long
 
+    if (executionTimeMs > timeoutMs && stage.context['markSuccessfulOnTimeout']) {
+      return new TaskResult(ExecutionStatus.SUCCEEDED)
+    }
```

Comment:
The `checkTimeout` method works by throwing an exception, which is no good. We could have the `checkTimeout` method not throw an exception, but we'd still have to do the same check to see if the stage has timed out after calling that method.

I considered moving the timeout checking logic into a separate method, but in the end, it would just be doing a `>` against the two numbers, so this seemed (to me) more straightforward.